### DISCUSSION TITLE
Operator % does not insert space

### DISCUSSION THREAD

**Discussion by LiShunGang** [microsoft/vscode-cpptools#8461] [Q&A]
This is my setting
![image](https://user-images.githubusercontent.com/36393104/143866248-aaa1e454-9c04-402b-99ca-c67915622c51.png)

After auto format, the space before "operator %" is removed. But this does not happen on other operators.
![image](https://user-images.githubusercontent.com/36393104/143866797-cfe0f7a3-9441-4d53-90ad-5e5aa60dd45a.png)

How to make it insert space near "operator %"?

Vscode version: 1.7.0
C/C++ version: 1.61.1

**Comment by sean-mcmanus** [comment]
It appears to be a limitation/bug with the vcFormat engine. You need to use something like "int i = ctx% m_op;" for the formatting to work. Is the % operator overloaded to have a side effect that modifies its arguments?

**Reply by LiShunGang** [reply]
> 

Yes, I have overloaded “operator %”. 'ctx' and 'm_op' are objects.  But as I can see, other binary operators work well...

**Reply by LiShunGang** [reply]
I got it!
This is affected by the “C_Cpp.vcFormat.space.pointerReferenceAlignment”.
Looks like a bug to me...

**Reply by bobbrow** [reply]
I wonder if that's because `^` and `%` are pointer/reference in managed C++.  We should check with VC folks and see if there's a setting to disable that for non-managed codebase formatting.
